woocommerce/woocommerce-ios
# Coupons: Remove redundant animation when reloading coupon list

Closes: #7135 


### Description

When trying to fix #6796, I introduced a new unique identifier to `TitleAndSubtitleAndStatusTableViewCell`'s view model to avoid sending duplicating items to the coupon list diffable table view data source. Since the identifier was of type `UUID`, every time a cell view model is created, it's a brand new one to the data source and therefore causes the list to animate when a new snapshot is applied.

This PR fixes the above issue with 2 parts:
- Make the cell view model identifier a string and use coupon IDs to construct the identifiers. This is to make sure that each cell is identified by the coupon it is displaying and stop the cells from animating when the list is reloaded.
- For some reason with the above change in place, the list still jumps a bit when reloaded. It looks like the offset of the list is always reset to show the top item. To fix this, I added `applySnapshotUsingReloadData` to reload the UI without animation. This function is only available since iOS 15, so the problem persists on iOS 14.

### Testing instructions

- Make sure that you have Coupon Management enabled in Settings > Experimental Features.
- Navigate to Menu > Coupons. If your list has more than 2 items, notice that the list doesn't jump when it is reloaded. Try pulling to refresh the list to confirm this.

### Screenshots


https://user-images.githubusercontent.com/5533851/175230562-d93509dc-ed0c-4b37-ace7-862356d7169c.mp4



---
